<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Preferences</class>
 <widget class="QDialog" name="Preferences">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>504</width>
    <height>675</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Preferences</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="lineWidth">
      <number>0</number>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>-38</y>
        <width>467</width>
        <height>662</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="0">
        <layout class="QVBoxLayout" name="verticalLayout_Prefs">
         <item>
          <widget class="QGroupBox" name="groupGeneral">
           <property name="title">
            <string>General</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QCheckBox" name="opt_checkUpdates">
              <property name="text">
               <string>Check for updates on program start</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="opt_intstantSearch">
              <property name="text">
               <string>Instant search when typing (library, history)</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="opt_reorder">
              <property name="text">
               <string>Automatically reorder circles after mouse release</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_9">
              <item>
               <widget class="QCheckBox" name="opt_zoom">
                <property name="text">
                 <string>Zoom circle when hovered, zoom factor:</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="opt_zoomFactor">
                <property name="maximumSize">
                 <size>
                  <width>50</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>1.2</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_8">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QCheckBox" name="opt_tooltip">
              <property name="text">
               <string>Show tooltip when hovered</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="opt_tooltipClipped">
              <property name="text">
               <string>Show tooltip if artist or title field is clipped</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_14">
              <item>
               <widget class="QLabel" name="label_6">
                <property name="text">
                 <string>Show tooltip if view zoom factor is less than: (0=disabled)</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="opt_tooltipViewZoom">
                <property name="maximumSize">
                 <size>
                  <width>50</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>0.5</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_13">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QCheckBox" name="opt_nowPlayingNotification">
              <property name="text">
               <string>Show &quot;Now playing&quot; notifications</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_10">
              <item>
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>Circle max string length</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="opt_maxStringLength">
                <property name="maximumSize">
                 <size>
                  <width>40</width>
                  <height>16777215</height>
                 </size>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_9">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_11">
              <item>
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>Circle max string length on hover</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="opt_maxStringLengthHover">
                <property name="maximumSize">
                 <size>
                  <width>40</width>
                  <height>16777215</height>
                 </size>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_10">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_13">
              <item>
               <widget class="QLabel" name="label_10">
                <property name="text">
                 <string>Autosave now playing playlist every x mins (0=disabled)</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="opt_autoSavePlaylistMins">
                <property name="maximumSize">
                 <size>
                  <width>40</width>
                  <height>16777215</height>
                 </size>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_12">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupGettingRelated">
           <property name="title">
            <string>Getting related tracks</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QCheckBox" name="opt_dontAddSameArtist">
                <property name="text">
                 <string>Don't add related tracks of the same artist</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_12">
              <item>
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>Don't add related track that has played in the last N hours: (0=disabled)</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="opt_dontAddHistoryLastNHours">
                <property name="maximumSize">
                 <size>
                  <width>50</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>24</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_11">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <property name="sizeConstraint">
               <enum>QLayout::SetDefaultConstraint</enum>
              </property>
              <item>
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>Number of related tracks to add per request: (0..10)</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="opt_relatedToAdd">
                <property name="minimumSize">
                 <size>
                  <width>30</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>30</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>5</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QLabel" name="label">
                <property name="text">
                 <string>Automatically get related if track children are less than: (0..3)</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="opt_getRelatedIfLessThan">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>30</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>1</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupLastFm">
           <property name="title">
            <string>Last.fm</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>Username:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="opt_lastFmUsername"/>
              </item>
              <item>
               <spacer name="horizontalSpacer_5">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="QLabel" name="label_4">
                <property name="text">
                 <string>Password:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="opt_lastFmPassword">
                <property name="echoMode">
                 <enum>QLineEdit::Password</enum>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_4">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_8">
              <item>
               <widget class="QPushButton" name="authenticateButton">
                <property name="text">
                 <string>Authenticate</string>
                </property>
                <property name="autoDefault">
                 <bool>false</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_5">
                <property name="text">
                 <string>Status:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="opt_lastFmStatus">
                <property name="text">
                 <string>...</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_7">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QCheckBox" name="opt_scrobble">
              <property name="text">
               <string>Scrobble</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="opt_nowPlaying">
              <property name="text">
               <string>Now Playing</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="opt_skipUnknownArtist">
              <property name="text">
               <string>Skip if artist field is &quot;Unkown Artist&quot;</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="0">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Expanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="buttonOk">
       <property name="text">
        <string>OK</string>
       </property>
       <property name="autoDefault">
        <bool>false</bool>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="butonCancel">
       <property name="text">
        <string>Cancel</string>
       </property>
       <property name="autoDefault">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonApply">
       <property name="text">
        <string>Apply</string>
       </property>
       <property name="autoDefault">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>opt_intstantSearch</tabstop>
  <tabstop>opt_reorder</tabstop>
  <tabstop>opt_zoomFactor</tabstop>
  <tabstop>opt_tooltip</tabstop>
  <tabstop>opt_dontAddSameArtist</tabstop>
  <tabstop>opt_relatedToAdd</tabstop>
  <tabstop>opt_getRelatedIfLessThan</tabstop>
  <tabstop>opt_lastFmUsername</tabstop>
  <tabstop>opt_lastFmPassword</tabstop>
  <tabstop>authenticateButton</tabstop>
  <tabstop>opt_scrobble</tabstop>
  <tabstop>opt_nowPlaying</tabstop>
  <tabstop>buttonOk</tabstop>
  <tabstop>butonCancel</tabstop>
  <tabstop>buttonApply</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonOk</sender>
   <signal>clicked()</signal>
   <receiver>Preferences</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>320</x>
     <y>632</y>
    </hint>
    <hint type="destinationlabel">
     <x>93</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>butonCancel</sender>
   <signal>clicked()</signal>
   <receiver>Preferences</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>401</x>
     <y>632</y>
    </hint>
    <hint type="destinationlabel">
     <x>37</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonApply</sender>
   <signal>clicked()</signal>
   <receiver>Preferences</receiver>
   <slot>apply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>482</x>
     <y>632</y>
    </hint>
    <hint type="destinationlabel">
     <x>128</x>
     <y>323</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>authenticateButton</sender>
   <signal>clicked()</signal>
   <receiver>Preferences</receiver>
   <slot>authenticate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>105</x>
     <y>519</y>
    </hint>
    <hint type="destinationlabel">
     <x>178</x>
     <y>340</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>apply()</slot>
  <slot>authenticate()</slot>
 </slots>
</ui>
